### PR TITLE
Contact endpoint

### PR DESCRIPTION
**Pull request by TheYisusGithub**

### CODE REVIEW

File: src/main/java/com/foundation/salesforce/core/api/ContactApi.java [hunk 1]
```diff
@@ -52,20 +51,20 @@ public static ContactApi getInstance() {
...
+     * @param contactBody parameter return the contact the bodu json.
      */
-    public void setContent(String Contactbody) {
-        restClientApi.buildSpec(Contactbody);
+    public void setContent(String contactBody) {
+        restClientApi.buildSpec(contactBody);
     }
 
     /**
      * Method let set task body that let build request.
```
Comment (titor-oopart) [inline src/main/java/com/foundation/salesforce/core/api/ContactApi.java]: change task for your endpoint

File: src/test/java/com/foundation/salesforce/hooks/ContactHooks.java [hunk 2]
```diff
@@ -45,8 +45,8 @@ public ContactHooks(Context context) {
      * Creates a Task before the scenarios.
      */
     @Before("@FindContact, @DeleteContact, @UpdateContact")
-    public void before_delete_task() {
-        String name = ValueAppender.prefix() + "Account" + ValueAppender.suffix();
+    public void beforeDeleteTask() {
```
Comment (titor-oopart) [inline src/test/java/com/foundation/salesforce/hooks/ContactHooks.java]: change the name accord to the endpoint

File: src/main/java/com/foundation/salesforce/core/api/ContactApi.java [hunk 3]
```diff
@@ -52,20 +51,20 @@ public static ContactApi getInstance() {
...
      *
-     * @param Contactbody parameter return the contact the bodu json.
+     * @param contactBody parameter return the contact the bodu json.
```
Comment (Cxristian) [inline src/main/java/com/foundation/salesforce/core/api/ContactApi.java]: what is bodu in the comment?

File: src/test/resources/features/contact/CreateContact.feature [hunk 4]
```diff
@@ -1,35 +1,35 @@
 Feature: Manage Tasks in Salesforce
```
Comment (Cxristian) [inline src/test/resources/features/contact/CreateContact.feature]: tag is missing

File: src/test/resources/schemas/ContactDeleteIDSchema.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+          "type": "string"
+        }
+      },
+      "required": [
+        "message",
+        "errorCode"
+      ]
+    }
+  ]
+}
```
Comment (Cxristian) [inline src/test/resources/schemas/ContactDeleteIDSchema.json]: add a new liine

File: src/test/java/com/foundation/salesforce/steps/ContactSteps.java [hunk 6]
```diff
@@ -66,70 +66,49 @@ public void userCreateNewContact(Map<String, String> inputContent) {
     /**
      * Method to let user sent a request.
      */
-    @When("User send de request post to contact endpoint")
+    @When("user send de request post to contact endpoint")
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactSteps.java]: it is not: user send de request
it is: user send the request

File: src/test/java/com/foundation/salesforce/steps/ContactSteps.java [hunk 7]
```diff
@@ -290,18 +258,28 @@ public void userUpdatesContactByIdFSA(String contactID) {
...
             Assert.assertEquals(context.getResponse().jsonPath().get(field.getKey()).toString(), field.getValue());
         }
     }
 
+    /**
+     * Method to let body fields.
+     *
+     * @param bodyFields parameter body fields.
+     */
     @And("In response we can found the following contain")
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactSteps.java:271]: I think this method has already been defined in common steps.
Please reuse it.

File: src/test/resources/features/contact/DeleteContact.feature [hunk 8]
```diff
@@ -1,20 +1,20 @@
 Feature: Manage Tasks in Salesforce
```
Comment (alszla) [inline src/test/resources/features/contact/DeleteContact.feature]: #1: As Cxristian just said, All of your feature files should begin with @Contact  as its first line.
#2: I guess you're not exactly managing tasks in this feature, are you?

File: src/test/resources/features/contact/FindContact.feature [hunk 9]
```diff
@@ -1,6 +1,6 @@
 Feature: Manage Tasks in Salesforce
```
Comment (alszla) [inline src/test/resources/features/contact/FindContact.feature]: Same as all feature files. That means: Tag at the beginning and get rid of that "Tasks" in the name of your features, please.

File: src/test/resources/schemas/ContactDeleteIDNotExitsSchema.json [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+          "type": "string"
+        }
+      },
+      "required": [
+        "errorCode",
+        "message"
+      ]
+    }
+  ]
+}
```
Comment (alszla) [inline src/test/resources/schemas/ContactDeleteIDNotExitsSchema.json]: Add a blank line.

File: src/test/java/com/foundation/salesforce/steps/ContactSteps.java [hunk 11]
```diff
@@ -290,18 +258,28 @@ public void userUpdatesContactByIdFSA(String contactID) {
...
+    @Given("user provides the following json")
     public void userProvidesTheFollowingJson(String jsonBodyString) {
         contactApi.setContent(jsonBodyString);
     }
 
+    /**
+     * Method to let body fields.
+     *
+     * @param bodyFields parameter body fields.
+     */
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactSteps.java:270]: This method is already defined in CommonSteps class.
Please reuse the methods defined there.
Besides, this and the next method are exactly the same, (if I am not wrong).

File: src/main/java/com/foundation/salesforce/core/api/ContactApi.java [hunk 12]
```diff
@@ -52,20 +51,20 @@ public static ContactApi getInstance() {
     /**
      * Method let set task body that let build request.
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/core/api/ContactApi.java]: Replace task for contact

File: src/test/resources/features/contact/CreateContact.feature (hunk 4 above)
Comment (alszla) [inline src/test/resources/features/contact/CreateContact.feature]: Hmmm, I guess you're not exactly managing Taks, are you?

File: src/test/resources/features/contact/UpdateContact.feature [hunk 13]
```diff
@@ -2,25 +2,25 @@ Feature: Manage Tasks in Salesforce
```
Comment (alszla) [inline src/test/resources/features/contact/UpdateContact.feature:3]: Add tag at the beginning and replace Tasks with the correct word.

**Review by Cxristian — APPROVED** [review]

**Review by alszla — APPROVED** [review]

**Pull request merged**